### ISSUE TITLE
Update to fo-dicom v4.0.0

### ISSUE THREAD

**Issue by Zaid-Safadi** [DICOMcloud/DICOMcloud#33]
## Expected Behavior
DICOMcloud uses latest version of fo-dicom v4.0.0

## Actual Behavior
Currently it is still on version v3.0.2.0

**Comment by shelbypereira** [comment]
I think this is becoming important, in fact the 4.0.2 version contains some breaking changes so would be nice to update to the latest version.

**Comment by shelbypereira** [comment]
I have forked and will make a pull request soon.  Many warnings to fix but following the recommendations in the obsolete method makes the fixes quick.  I have fixed the warnings witht the name changes of certain methods.  Therare a few obesolete messages associated with the updateing of image data.  These require a bit more care to fix, but the answer over here https://stackoverflow.com/questions/53096584/how-to-change-the-pixel-data-of-dicom-file-using-fo-dicom
shows the solution.

**Comment by Zaid-Safadi** [comment]
@shelbypereira you are welcome to submit a PR and it is very much appreciated! I meant to do this for a while now but unfortunately didn't have the time. 

I can help with testing and if specific questions comes up in fo-dicom I'll be happy to answer to my knowledge.

Thank you!

**Comment by shelbypereira** [comment]
Great thanks for your feedback.  I have 3 more warnings to fix,  related to the image array.  I just wanted to highlight one change where there was an ambiguitiy In the file JpegMediaWriter.cs line 52:

      
      -      var bitmap = dicomImage.RenderImage (frameIndex).AsBitmap();
       +     var bitmap = dicomImage.RenderImage (frameIndex).AsSharedBitmap();
            var stream = new MemoryStream ( ) ;
            
            bitmap.Save (stream, System.Drawing.Imaging.ImageFormat.Jpeg );


I used AsSharedBitmap here (another choice was AsClonedBitmap)  As the image is being saved immediately and the function returns after the same,  I think the clone is not needed here and may be slow for large images.  However I mam not sure whether the original AsBitmap makes a clone or not.

All other changes up to now had noambiguity there was only one logical fix to remove the obsolete warning.

**Comment by Zaid-Safadi** [comment]
@shelbypereira `AsSharedBitmap` seems to be the right choice here. However not sure if there are any complications of using it over the `AsClonedBitmap`. I'll try to do some digging.

**Comment by shelbypereira** [comment]
I have been testing more.  I installed DICOMCloud/DICOMWebjs  and I see that exceptions are being thrown (not sure they were there before the migration,  but I am assuming no.  I any case to prove things are working I think at least the following tests are needed:

1 test for each of the supported web apis (QidoRS, StowRS, WadoRS, WadoUri)
2. 1 test with an 8 bit image, another with a 16 bit image.  to confirm VR is ok, but  I suspect it works as expected based on feedback from the fo-dicom team.

**Comment by Zaid-Safadi** [comment]
@shelbypereira I tried your update and I see the error when you query. I remember hitting this same error when I first tried to upgrade but didn't figure it out at the time. 
The query initializes a default empty dataset as a template. 
`studyDs.Add<object>(fo.DicomTag.SpecificCharacterSet,null) ;`
in v4 I get an exception when trying something similar:
`studyDs.Add<object>(fo.DicomTag.SpecificCharacterSet,(object)null) ;`
with the exception message:

> InvalidOperationException: Unable to create DICOM element of type CS with values of type System.Object

I posted this to the fo-dicom group and hope to get some help.
